__std__/nix/plutus/devshells/plutus-shell.nix

```diff
@@ -9,11 +9,27 @@ inputs.std.std.lib.mkShell {
   name = "plutus-shell";
 
   imports = [
+    cell.library.plutus-project.devshell
     inputs.cells.toolchain.devshellProfiles.haskell
   ];
 
-  packages = pkgs.lib.optionals (!pkgs.stdenv.isDarwin) [
+  packages = [
+    pkgs.libsodium-vrf
+    pkgs.libsodium-vrf.dev
+    pkgs.secp256k1
+  ] ++ pkgs.lib.optionals (!pkgs.stdenv.isDarwin) [
     pkgs.rPackages.plotly
     pkgs.R
   ];
+
+  env = [
```

Comment:
Why do we need to do this? I would expect the devshell stuff to manage its own environment variables like this? Really needs a comment at least.

Reply:
I was also expecting the [devshell](https://github.com/numtide/devshell) to do this automatically.

Reply:
devshell want all environment variables to be explicit. There is some limited support to ease this if we import https://github.com/numtide/devshell/blob/master/extra/language/c.nix though.

Reply:
So I've tested the new shell and I can do a full `cabal build all` on my machine.

Following the `c.nix` example you posted, I reckon the `env` stuff should therefore be moved to 
`inputs.cells.toolchain.devshellProfiles.haskell` instead?

Reply:
ah yeah. But could probably also be upstreamed in https://github.com/input-output-hk/haskell.nix/blob/7462e537efccfc4471908a5c96c5c72b2646f985/lib/project-overlays.nix#L16
Also working a patch to avoid explicitly listing libsodium/secp256k1.

Reply:
Are we the first people to use this? :sweat_smile: